lms/djangoapps/discussion_api/api.py

```diff
@@ -415,7 +415,7 @@ def get_comment_list(request, thread_id, endorsed, page, page_size, mark_as_read
         raise Http404
     num_pages = (resp_total + page_size - 1) / page_size if resp_total else 1
 
-    results = [CommentSerializer(response, context=context).data for response in responses]
+    results = [CommentSerializer(response, remove_fields=["children"], context=context).data for response in responses]
```

Comment:
@wajeeha-khalid Since we have now updated our version of DRF, can you look into using the `exclude` option?  If that works, you can also replace the other usage of `remove_fields` in the discussion API.